config/src/main/java/com/epam/aidial/core/config/validation/ConformToMetaSchemaValidator.java

```diff
@@ -0,0 +1,36 @@
+package com.epam.aidial.core.config.validation;
+
+import com.epam.aidial.core.metaschemas.MetaSchemaHolder;
+import com.networknt.schema.InputFormat;
+import com.networknt.schema.JsonSchema;
+import com.networknt.schema.JsonSchemaFactory;
+import com.networknt.schema.SpecVersion;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+import java.util.Map;
+
+public class ConformToMetaSchemaValidator implements ConstraintValidator<ConformToMetaSchema, Map<String, String>> {
+
+    private static final JsonSchemaFactory SCHEMA_FACTORY = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V7);
+    private static final JsonSchema SCHEMA = SCHEMA_FACTORY.getSchema(MetaSchemaHolder.getCustomApplicationMetaSchema());
+
+    @Override
+    public boolean isValid(Map<String, String> stringStringMap, ConstraintValidatorContext context) {
+        if (stringStringMap == null) {
```

Comment:
Does this method expect nulls? I would dropped an exception in this case or removed this piece of code completely.

Reply:
null allowance should be controlled by separated constraint and validator.